apps/fishing-map/features/datasets/upload/NewDataset.tsx

```diff
@@ -272,10 +265,12 @@ function NewDataset(): React.ReactElement {
   const onConfirmClick = useCallback(
     async (dataset: DatasetMetadata, datasetFile?: File) => {
       if (dataset) {
-        const { fields, ...meta } = dataset as DatasetMetadata
         const { payload, error: createDatasetError } = await dispatchUpsertDataset({
           dataset: {
-            ...meta,
+            ...dataset,
+            fieldsAllowed: dataset.configuration?.configurationUI?.filter
```

Comment:
is needed to duplicate the state here? I mean, why not using directly the `fieldsAllowed` property in the dataset instead of the `filter`

Reply:
We need the newly set `filter` to update the fieldsAllowed, if we do not have `filter` then yes, we would need to use already set dataset `filedAllowed`